catalog/src/main/java/io/kaoto/backend/api/metadata/catalog/StepCatalog.java

```diff
@@ -64,52 +64,59 @@ private boolean isClusterAvailable() {
     private void addCluster(final List<ParseCatalog<Step>> catalogs,
                             final boolean clusterAvailable) {
         if (clusterAvailable) {
-            stepCatalogParsers.stream().parallel().forEach(parser -> catalogs.add(parser.getParserFromCluster()));
```

Comment:
when using parallel stream, remember that  each processing step may run in a different thread and sometime truly parallel depending on the hardware hence, calls to things like `catalogs.add` must either performed against a thread safe collection or concurrent access must be guarded somehow or collected for further processing otherwise the behavior is undefined (in this case, as we have moved to an `ArrayList` it often manifests with an `ArrayOutOfBound` exception before, it was likely causing some cryptic errors, loss of data and flaky tests).